packages/features/bookings/Booker/components/AvailableTimeSlots.tsx

```diff
@@ -34,30 +44,30 @@ export const AvailableTimeSlots = ({ extraDays, limitHeight, seatsPerTimeslot }:
   };
 
   const schedule = useScheduleForEvent({
-    prefetchNextMonth: !!extraDays && dayjs(date).month() !== dayjs(date).add(extraDays, "day").month(),
+    prefetchNextMonth,
+    monthCount,
   });
+  const nonEmptyScheduleDays = useNonEmptyScheduleDays(schedule?.data?.slots);
 
   // Creates an array of dates to fetch slots for.
   // If `extraDays` is passed in, we will extend the array with the next `extraDays` days.
   const dates = useMemo(
     () =>
       !extraDays
         ? [date]
-        : [
-            // If NO date is selected yet, we show by default the upcomming `nextDays` days.
-            date,
-            ...Array.from({ length: extraDays }).map((_, index) =>
-              dayjs(date)
-                .add(index + 1, "day")
-                .format("YYYY-MM-DD")
-            ),
-          ],
+        : nonEmptyScheduleDays.length > 0 
+          ? nonEmptyScheduleDays.slice(sliceFrom, sliceTo)
+          :[],
     [date, extraDays]
   );
-
-  const isMultipleDates = dates.length > 1;
-  const slotsPerDay = useSlotsForMultipleDates(dates, schedule?.data?.slots);
-
+  
+  const monthViewSlots = schedule?.data?.slots  && nonEmptyScheduleDays.includes(date) ? Object.values(schedule.data.slots).slice(nonEmptyScheduleDays.indexOf(date), nonEmptyScheduleDays.indexOf(date) + 1) : [];
+  const slotsPerDay = schedule?.data?.slots 
+    ? useSlotsForAvailableDates(dates, isColumnView 
+      ? Object.values(schedule.data.slots).slice(sliceFrom, sliceTo) 
+      : monthViewSlots ) 
+    : useSlotsForAvailableDates(dates, []);
+  
```

Comment:
you can't use useSlotsForAvailableDates hook conditonally
<img width="801" alt="Screenshot 2023-07-03 at 3 28 29 PM" src="https://github.com/calcom/cal.com/assets/53316345/16952552-8776-4990-9006-cfcb76729e1b">

Reply:
Ok let me add all the conditions in hook itself.

Reply:
Updated dude